src/Nethermind/Nethermind.HealthChecks/HealthChecksPlugin.cs

```diff
@@ -149,7 +152,7 @@ public Task InitRpcModules()
                 if (_logger.IsInfo) _logger.Info("Health RPC Module has been enabled");
             }
 
-            if (_api.SpecProvider!.TerminalTotalDifficulty is not null)
+            if (_mergeConfig.Enabled && _api.SpecProvider!.TerminalTotalDifficulty is not null)
```

Comment:
Hmmm Merge.Enabled = false with TerminalTotalDifficulty **NOT NULL** sounds like configuration issue